src/ck-core/cklocation.C

```diff
@@ -2536,6 +2536,11 @@ int CkLocMgr::deliverMsg(CkArrayMessage *msg, CkArrayID mgr, CmiUInt8 id, const
       }
 #endif
       msg->array_hops()++;
+      // If we are hopping more than twice, we've discovered a stale chain
+      // of cache entries. Just route through home instead.
+      if (msg->array_hops() > 2 && CkMyPe() != homePe(id)) {
+        destPE = homePe(id);
```

Comment:
Is there a lookup cost of calling `homePe()`? Is it worth making a local variable to only call it once?

Reply:
The cost is (or should be) very small but it depends on the mapping used. If it's made a local variable then it will always be called once, whereas in this case it is never called (due to short circuiting) unless we are detecting a long chain.